### ISSUE TITLE
Channel and User Group Flags (Icons)

### ISSUE THREAD

**Issue by Xylemon** [mumble-voip/mumble#1632]
It would be nice if you could give channels and user groups their own icons (this was also reported on the SourceForge tracker, just putting it here as well).

**Comment by bontibon** [comment]
Could you do a quick mockup of where you think the icons should appear? I am wondering if there would be confusion between a user's group icon and the status icons (e.g. muted, authenticated, etc.).

**Comment by Xylemon** [comment]
Here's a quick mockup I did in Gimp:

http://www.bloodbathsoftworks.com/xylemon/images/screencaps/mumbleiconmockup.png

In this, JohnBoy and I are both part of the "Linux group". However he's only a guest, I have myself a custom Admin icon. The other channels have their own icons as well, such as Recording Studio with a mic.

**Comment by bontibon** [comment]
Do you think maybe the groups icons in the mockup could be easily confused as status icons. Maybe putting the icons before or after the user/channel name would be better:

After:
![after](https://cloud.githubusercontent.com/assets/110883/7256217/3b306cbc-e824-11e4-9efe-a2448ec3d1cb.png)

Before:
![before](https://cloud.githubusercontent.com/assets/110883/7256293/9f2277ce-e824-11e4-97da-9f247b092eda.png)

**Comment by Xylemon** [comment]
The mock up I did was similar to TeamSpeak 3, however the "Before name" mock up you did could work as well.

**Comment by bheart** [comment]
I've actually been working on this in the past few days, though in my implementation the icons are simply prepended to the status icons (registered, suppressed etc.)

Is it necessarily a bad thing to leave custom flags in with status icons? The point of both types is to describe certain traits of the user, no reason to separate them imo.

**Comment by bendem** [comment]
I'd personally prefer having all icons in one place as well. Maybe it could be configurable?

**Comment by Xylemon** [comment]
I too would like to see them configurable (for I like the way TS3 does it).

@bheart would your implementation ultimately allow custom icons for groups?

**Comment by bheart** [comment]
@Xylemon probably, that's why I started working on it in the first place.

How do you imagine the UI for setting/uploading icons? You need to be able to set them on a per-user, group and channel basis, I'm thinking.

**Comment by mkrautz** [comment]
The complication I see in this is that there's no such thing as a server-wide "group".

A group can by set on the root channel, and sub-channels can inherit that group.  But they can also decide not to, and set its list of users for that group.  A group in a sub channel can also exclude inherited users, add new ones, and so on.

So... Is this thing a separate concept?  Or do we just restrict you to only be able to set icons for global groups (groups that are defined on the root channel), and that icon will be used for all groups with the same name in sub-channels, regardless of whether the set of members is the same?

**Comment by bendem** [comment]
Why not apply a picture per group be it a root group or not and only display it if the group applies in the channel the user is in?

**Comment by Kissaki** [comment]
Could you please link the SF FR you mentioned?

And mention your use-case if those are not listed there yet.

In the mockup images you posted, it seems confusing to me.

**Comment by bontibon** [comment]
@Kissaki http://sourceforge.net/p/mumble/feature-requests/195/

**Comment by bheart** [comment]
@Kissaki like @Xylemon said, icons make for a great, concise way of describing a certain user's role on the server or the related community. Imagine you had a question for an administrator, scanning the tree for an icon resembling, well, an icon an administrator might have, is much friendlier than asking around or looking for a list of staff.

**Comment by Xylemon** [comment]
Plus, I feel it gives the server more character and visual understanding when a channel has an icon associated with it. For instance, you have a Counter-Strike channel with the icon of the game, you can already know to go there without even reading the full channel name.

**Comment by Xylemon** [comment]
@bheart did you ever finish your implementation of this?

**Comment by bheart** [comment]
@Xylemon I forgot about it after a while, [this was as far as I got](https://github.com/mumble-voip/mumble/compare/mumble-voip:84a8bbc...bheart:1da0184).

**Comment by Keridos** [comment]
Would love to see this implemented.

**Comment by Keridos** [comment]
@bheart Do you intend of finishing that sometimes in the future? If not, how can I properly setup a dev environment in which to write that code and test it properly on Windows?

**Comment by bheart** [comment]
@Keridos I wouldn't count on it, but I might revisit it some time. As for building on Windows, here's a guide http://wiki.mumble.info/wiki/BuildingWindows - I'm not familiar with the process

**Comment by nikkelj** [comment]
+1, this is needed, i see a lot of people hate mumble because it looks terrible out of the box

**Comment by Keridos** [comment]
Yeah, currently Mumble in my opinion needs a more attractive Interface, partially the 1.3.0 Release might do it with the better Banlist and Userlist configuration. But things like locked registration names for non admins and missing icons (no way to see who is an admin on the server unless explicitly stated somewhere else f.ex.) may be the two biggest turn off for new Mumble Users.

**Comment by TBotV63** [comment]
I personally think the dark theme looks gorgeous and it's all pretty usable imho. (I'm on 1.3.0 snapshot)
This is the only thing I'm really missing. As for fixed registration names, I personally even consider that a feature. After all, you want be able to recognise people. But that should be discussed somewhere else as it's not related to this particular issue.

**Comment by preterive** [comment]
Would love to see this implemented in the 1.3

**Comment by preterive** [comment]
I wish that i can set a icon (for example 16x16) for every group Like in ts3

**Comment by bendem** [comment]
the best thing you can do to get this feature more attention is to vote for it (click the thumbs up icon on the first comment).

**Comment by preterive** [comment]
I don't See any icon...

**Comment by Keridos** [comment]
This is already a pretty well discussed feature request. And it got a few upvotes. Why hesitate so much on this?

**Comment by bontibon** [comment]
@Keridos https://www.bountysource.com/issues/11644687-channel-and-user-group-flags-icons

**Comment by bendem** [comment]
@Keridos feel free to write a patch. 😉

**Comment by Keridos** [comment]
I'd have to get into mumble and c++ at the same time :) So far I can only write C, Java and a few minor languages.

**Comment by anure** [comment]
Can we get this plz team, I believe in you. Hercules 👏 Hercules 👏 Hercules 👏

**Comment by AiyionPrime** [comment]
I'd love to see this in mumble. It would give the UI way more overview in who to ask something.

**Comment by Krzmbrzl** [comment]
@AiyionPrime users do have a dedicated icon by now. Doesn't that suffice?
What is still missing? :)

**Comment by AiyionPrime** [comment]
I got a mention with this text, but Github doesn't show it:

> @AiyionPrime users do have a dedicated icon by now. Doesn't that suffice?
What is still missing? :)

That users can have an Icon is a neat feature, but quite useless; tbh.
The greater worth would come from icons/or badges, that are shown, if you are part of a specific group.
E.g. an icon, that all the mods of a server get, another that all Clan-Leaders have, ...
I thought this was meant by "User Group Flags".
In combination with Channel Icons (shown next to the afk channel in the example above) this would be quite useful, as a channel could be marked with the goups icons, that are allowed to enter, if it's not open for all.
This would grant a fast overview, which places are meant for which users.

If this is already a thing in mumble, I'm sorry and would like to know, where to configure them.
Thanks!

**Comment by Krzmbrzl** [comment]
> I got a mention with this text, but Github doesn't show it:

This due to me having realized that I misinterpreted the issue's title and thus having deleted my comment :)

You are correct that this wouold be a cool feature to have. I initially thought this issue was about not being able to differentiate between channels and users due to missing icons (thus my comment). Somehow I skipped the "group" part in the title ^^

> If this is already a thing in mumble, I'm sorry and would like to know, where to configure them.

Nope it is not.

> channel could be marked with the goups icons, that are allowed to enter, if it's not open for all.

Not the same but for this specific use-case I just recently committed #3929. Maybe that'll do in the meantime :)

**Comment by AiyionPrime** [comment]
#3929 is quite nice.
It does not grant the ability to show the user, which groups he is lacking to enter the the channel, though, which this idea would.

Thanks for clarifying!

**Comment by jonpas** [comment]
Group flags/icons would be neat! Especially with recent plugin API developments and uses that come from it - visible hierarchy is useful in a lot of communities.
